File: equipment-tracker-frontend/src/app/services/transaction.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Transaction } from '../models/transaction.model'; // Define this model

@Injectable({
  providedIn: 'root'
})
export class TransactionService {
  private apiUrl = 'http://localhost:8080/api/transaction';

  constructor(private http: HttpClient) { }

  getAllTransactions(): Observable<Transaction[]> {
    return this.http.get<Transaction[]>(this.apiUrl);
  }

  getTransactionsByUserId(userId: number): Observable<Transaction[]> {
    return this.http.get<Transaction[]>(`${this.apiUrl}/user/${userId}`);
  }

  getTransactionsByUsername(username: string): Observable<Transaction[]> {
    return this.http.get<Transaction[]>(`${this.apiUrl}?username=${username}`);
  }

  createTransaction(transaction: Transaction): Observable<Transaction> {
    return this.http.post<Transaction>(this.apiUrl, transaction);
  }
}

File: src/main/java/com/roehr/equipmenttracker/service/TransactionService.java
package com.roehr.equipmenttracker.service;

import com.roehr.equipmenttracker.model.Transaction;
import java.util.List;

public interface TransactionService {

    List<Transaction> getTransactionsByUserId(Long userId);

    Transaction save(Transaction transaction);

    List<Transaction> getAllTransactions();

    Transaction findById(Long transactionId);
}
